stdlib/list.mli

```diff
@@ -491,6 +491,21 @@ val split : ('a * 'b) list -> 'a list * 'b list
    Not tail-recursive.
  *)
 
+val split_with : ('c -> 'a * 'b) -> 'c list -> 'a list * 'b list
+(** [split_with f l] returns a returns a pair of lists [(l1, l2)] such that,
+    for each element [x] of the input list [l], if [f x] is [(y1, y2)] then
+    [y] is in [l1] and [y2] is in [l2].
+
+    The output elements are included in [l1] and [l2] in the same
+    relative order as the corresponding input elements in [l].
```

Comment:
I find the explanation a bit long-winded (even if precise). An possible alternative:
```
[split_map f l] is equivalent to [split (map f l)] but avoids allocating intermediate lists.
```